zebrad/src/components/inbound/tests/real_peer_set.rs

```diff
@@ -0,0 +1,498 @@
+/// by querying the inbound service via a local TCP connection.
+///
+/// Uses a real Zebra network stack with a local listener address,
+/// and an isolated Zebra inbound TCP connection.
+#[tokio::test]
+async fn inbound_peers_empty_address_book() -> Result<(), crate::BoxError> {
+    let (
+        // real services
+        connected_peer_service,
+        inbound_service,
+        _peer_set,
+        _mempool_service,
+        _state_service,
+        // mocked services
+        _mock_block_verifier,
+        _mock_tx_verifier,
+        // real tasks
+        block_gossip_task_handle,
+        tx_gossip_task_handle,
+        // real open socket addresses
+        listen_addr,
+    ) = setup().await;
+
+    // Use inbound directly
+    let request = inbound_service.clone().oneshot(Request::Peers);
+    let response = request.await;
+    match response.as_ref() {
+        Ok(Response::Peers(single_peer)) if single_peer.len() == 1 => {
+            assert_eq!(single_peer.first().unwrap().addr(), listen_addr)
+        }
+        Ok(Response::Peers(_peer_list)) => unreachable!(
+            "`Peers` response should contain a single peer, \
+             actual result: {:?}",
+            response
+        ),
+        _ => unreachable!(
+            "`Peers` requests should always respond `Ok(Response::Peers(_))`, \
+             actual result: {:?}",
+            response
+        ),
+    };
+
+    // Use the connected peer via a local TCP connection
+    let request = connected_peer_service.clone().oneshot(Request::Peers);
+    let response = request.await;
+    match response.as_ref() {
+        Ok(Response::Peers(single_peer)) if single_peer.len() == 1 => {
+            assert_eq!(single_peer.first().unwrap().addr(), listen_addr)
+        }
+        Ok(Response::Peers(_peer_list)) => unreachable!(
+            "`Peers` response should contain a single peer, \
+             actual result: {:?}",
+            response
+        ),
+        _ => unreachable!(
+            "`Peers` requests should always respond `Ok(Response::Peers(_))`, \
+             actual result: {:?}",
+            response
+        ),
+    };
+
+    let block_gossip_result = block_gossip_task_handle.now_or_never();
+    assert!(
+        matches!(block_gossip_result, None),
+        "unexpected error or panic in block gossip task: {:?}",
+        block_gossip_result,
+    );
+
+    let tx_gossip_result = tx_gossip_task_handle.now_or_never();
+    assert!(
+        matches!(tx_gossip_result, None),
+        "unexpected error or panic in transaction gossip task: {:?}",
+        tx_gossip_result,
+    );
+
+    Ok(())
+}
+
+/// Check that a network stack with an empty state responds to block requests with `notfound`.
+///
+/// Uses a real Zebra network stack, with an isolated Zebra inbound TCP connection.
+#[tokio::test]
+async fn inbound_block_empty_state_notfound() -> Result<(), crate::BoxError> {
```

Comment:
Minor: Same as above.

```suggestion
async fn inbound_block_empty_state_notfound() {
```